sdks/go/pkg/beam/registration/emitterIterRegistration.go

```diff
@@ -0,0 +1,298 @@
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/typex"
+)
+
+type emitNative1[T any] struct {
+	n  exec.ElementProcessor
+	fn interface{}
+
+	ctx   context.Context
+	ws    []typex.Window
+	et    typex.EventTime
+	value exec.FullValue
+}
+
+func (e *emitNative1[T]) Init(ctx context.Context, ws []typex.Window, et typex.EventTime) error {
+	e.ctx = ctx
+	e.ws = ws
+	e.et = et
+	return nil
+}
+
+func (e *emitNative1[T]) Value() interface{} {
+	return e.fn
+}
+
+func (e *emitNative1[T]) invoke(val T) {
+	e.value = exec.FullValue{Windows: e.ws, Timestamp: e.et, Elm: val}
+	if err := e.n.ProcessElement(e.ctx, &e.value); err != nil {
+		panic(err)
+	}
+}
+
+type emitNative2[T1, T2 any] struct {
+	n  exec.ElementProcessor
+	fn interface{}
+
+	ctx   context.Context
+	ws    []typex.Window
+	et    typex.EventTime
+	value exec.FullValue
+}
+
+func (e *emitNative2[T1, T2]) Init(ctx context.Context, ws []typex.Window, et typex.EventTime) error {
+	e.ctx = ctx
+	e.ws = ws
+	e.et = et
+	return nil
+}
+
+func (e *emitNative2[T1, T2]) Value() interface{} {
+	return e.fn
+}
+
+func (e *emitNative2[T1, T2]) invoke(key T1, val T2) {
+	e.value = exec.FullValue{Windows: e.ws, Timestamp: e.et, Elm: key, Elm2: val}
+	if err := e.n.ProcessElement(e.ctx, &e.value); err != nil {
+		panic(err)
+	}
+}
+
+type emitNative1WithTimestamp[T any] struct {
+	n  exec.ElementProcessor
+	fn interface{}
+
+	ctx   context.Context
+	ws    []typex.Window
+	et    typex.EventTime
+	value exec.FullValue
+}
+
+func (e *emitNative1WithTimestamp[T]) Init(ctx context.Context, ws []typex.Window, et typex.EventTime) error {
+	e.ctx = ctx
+	e.ws = ws
+	e.et = et
+	return nil
+}
+
+func (e *emitNative1WithTimestamp[T]) Value() interface{} {
+	return e.fn
+}
+
+func (e *emitNative1WithTimestamp[T]) invoke(et typex.EventTime, val T) {
+	e.value = exec.FullValue{Windows: e.ws, Timestamp: et, Elm: val}
+	if err := e.n.ProcessElement(e.ctx, &e.value); err != nil {
+		panic(err)
+	}
+}
+
+type emitNative2WithTimestamp[T1, T2 any] struct {
+	n  exec.ElementProcessor
+	fn interface{}
+
+	ctx   context.Context
+	ws    []typex.Window
+	et    typex.EventTime
+	value exec.FullValue
+}
+
+func (e *emitNative2WithTimestamp[T1, T2]) Init(ctx context.Context, ws []typex.Window, et typex.EventTime) error {
+	e.ctx = ctx
+	e.ws = ws
+	e.et = et
+	return nil
+}
+
+func (e *emitNative2WithTimestamp[T1, T2]) Value() interface{} {
+	return e.fn
+}
+
+func (e *emitNative2WithTimestamp[T1, T2]) invoke(et typex.EventTime, key T1, val T2) {
+	e.value = exec.FullValue{Windows: e.ws, Timestamp: et, Elm: key, Elm2: val}
+	if err := e.n.ProcessElement(e.ctx, &e.value); err != nil {
+		panic(err)
+	}
+}
+
+// RegisterEmitter1 registers parameters from your DoFn with a
+// signature func(T) and optimizes their execution.
+// This must be done by passing in a reference to an instantiated version
+// of the function, aka:
+// registration.RegisterEmitter1[T]((*func(T))(nil))
+func RegisterEmitter1[T1 any](e *func(T1)) {
```

Comment:
Please drop the Register prefix for all the functions here. As per your example users will already have  `registration` as a prefix. `registration.Emitter1` is more readable. 

Hmmm, it's also not too late to just call it the `register` package too... but that's definitely a change for another PR.

I suspect this is hangover from when it was in the `beam` package.

Reply:
> I suspect this is hangover from when it was in the beam package.

You are correct :) Fixed

> Hmmm, it's also not too late to just call it the register package too... but that's definitely a change for another PR.

I like that more and will follow up with that change once in flight prs are in.